### PR TITLE
Test generalization - now configurable via environment variables or dotenv

### PR DESCRIPTION
**Pull request by fjrojasgarcia**
This PR reference to #48 with a renamed branch according with the specific changes done.
You can see the conversation and comments there.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,7 +27,11 @@ install:
   - mkdir test/data/.qtum
 
 script:
-  - cd qtum-0.14.13/bin; ./qtumd -testnet -logevents -rpcuser=bodhi -rpcpassword=bodhi -datadir=/home/travis/build/bodhiproject/qweb3.js/test/data/.qtum & 
+  - pwd
+  - ls -l
+  #- cd qtum-0.14.13/bin; ./qtumd -testnet -logevents -rpcuser=bodhi -rpcpassword=bodhi -datadir=/home/travis/build/bodhiproject/qweb3.js/test/data/.qtum & 
+  # - cd qtum-0.14.13/bin; ./qtumd -testnet -logevents -rpcuser=bodhi -rpcpassword=bodhi -datadir=$TRAVIS_BUILD_DIR/qweb3.js/test/data/.qtum &
```
Comment (dwalintukan) [inline .travis.yml]: good refactor 👍 
remove 2 commented lines

File: test/config/config.js [hunk 2]
```diff
@@ -5,4 +5,35 @@ const Config = {
...
+  } else {
+    return qtumRPCAddress;
+  }
+};
+
+module.exports = {  
+  Config: Config,
+  getQtumRPCAddress: getQtumRPCAddress,
+  getDefaultQtumAddress: getDefaultQtumAddress  
+}
```
Comment (dwalintukan) [inline test/config/config.js]: add extra line at end of file (EOF)

File: test/config/config.js [hunk 3]
```diff
@@ -5,4 +5,35 @@ const Config = {
   ENCRYPTED_WALLET_TESTS: false,
 };
 
-module.exports = Config;
+// Load environment variables
+require('dotenv').config();
+var qtumRPCAddress = "QTUM_RPC_ADDRESS" in process.env ? String(new Buffer(process.env["QTUM_RPC_ADDRESS"])) : undefined;
+var qtumAddress = "SENDER_ADDRESS" in process.env ? String(new Buffer(process.env["SENDER_ADDRESS"])) : undefined;
```
Comment (dwalintukan) [inline test/config/config.js]: change both var to `const`

File: test/config/config.js [hunk 4]
```diff
@@ -5,4 +5,35 @@ const Config = {
   ENCRYPTED_WALLET_TESTS: false,
 };
 
-module.exports = Config;
+// Load environment variables
+require('dotenv').config();
```
Comment (dwalintukan) [inline test/config/config.js]: move to top of file with other imports

File: test/config/config.js [hunk 5]
```diff
@@ -5,4 +5,35 @@ const Config = {
...
+    return Config.QTUM_RPC_ADDRESS;
+  } else {
+    return qtumRPCAddress;
+  }
+};
+
+module.exports = {  
+  Config: Config,
+  getQtumRPCAddress: getQtumRPCAddress,
+  getDefaultQtumAddress: getDefaultQtumAddress
```
Comment (dwalintukan) [inline test/config/config.js]: can use shortform:
```
module.exports = {
  Config,
  getQtumRPCAddress,
  getDefaultQtumAddress,
}
```

File: test/contract.js [hunk 6]
```diff
@@ -5,7 +5,9 @@ const chai = require('chai');
 
 const assert = chai.assert;
 
-const Config = require('./config/config');
+const config = require('./config/config');
+const Config = config.Config;
```
Comment (dwalintukan) [inline test/contract.js]: change to `const { Config } = require('./config/config');`

File: test/qweb3.js [hunk 7]
```diff
@@ -4,17 +4,25 @@ const chai = require('chai');
...
 const expect = chai.expect;
 
+const bs58 = require('bs58')
+
 const Qweb3 = require('../src/qweb3');
 const Formatter = require('../src/formatter');
-const Config = require('./config/config');
+
+const config = require('./config/config');
+const Config = config.Config;
```
Comment (dwalintukan) [inline test/qweb3.js]: `const { Config, getQtumRPCAddress, getDefaultQtumAddress  } = require('./config/config');`

File: test/qweb3.js [hunk 8]
```diff
@@ -4,17 +4,25 @@ const chai = require('chai');
...
+
+console.log(`Your Qtum RPC address is ${config.getQtumRPCAddress()}`);
+console.log(`Your Default Qtum address is ${config.getDefaultQtumAddress()}`);
+
 const ContractMetadata = require('./data/contract_metadata');
 
 describe('Qweb3', () => {
-  const QTUM_ADDRESS = 'qKjn4fStBaAtwGiwueJf9qFxgpbAvf1xAy';
+  const QTUM_ADDRESS = config.getDefaultQtumAddress();
```
Comment (dwalintukan) [inline test/qweb3.js]: after above import change -> `getDefaultQtumAddress();`

File: test/qweb3.js [hunk 9]
```diff
@@ -4,17 +4,25 @@ const chai = require('chai');
...
   let qweb3;
 
   beforeEach(() => {
-    qweb3 = new Qweb3(Config.QTUM_RPC_ADDRESS);
+    qweb3 = new Qweb3(config.getQtumRPCAddress());
```
Comment (dwalintukan) [inline test/qweb3.js]: after above import change `new Qweb3(getQtumRPCAddress());`

File: test/qweb3.js [hunk 10]
```diff
@@ -289,13 +297,19 @@ describe('Qweb3', () => {
   /** ******** RAW TRANSACTIONS ********* */
   describe('getHexAddress()', () => {
     it('returns the hex address', async () => {
-      assert.equal(await qweb3.getHexAddress(QTUM_ADDRESS), '17e7888aa7412a735f336d2f6d784caefabb6fa3');
+      const hexDecodedaAddress = bs58.decode(QTUM_ADDRESS).toString('hex')
```
Comment (dwalintukan) [inline test/qweb3.js]: `hexDecodedAddress`, typo

File: test/qweb3.js [hunk 11]
```diff
@@ -4,17 +4,25 @@ const chai = require('chai');
...
+const bs58 = require('bs58')
+
 const Qweb3 = require('../src/qweb3');
 const Formatter = require('../src/formatter');
-const Config = require('./config/config');
+
+const config = require('./config/config');
+const Config = config.Config;
+
+console.log(`Your Qtum RPC address is ${config.getQtumRPCAddress()}`);
```
Comment (dwalintukan) [inline test/qweb3.js]: after above import change: `${getQtumRPCAddress()}`

File: test/qweb3.js [hunk 12]
```diff
@@ -4,17 +4,25 @@ const chai = require('chai');
...
+console.log(`Your Default Qtum address is ${config.getDefaultQtumAddress()}`);
```
Comment (dwalintukan) [inline test/qweb3.js]: after above import change: `${getDefaultQtumAddress()}`

File: .travis.yml (hunk 1 above)
Comment (fjrojasgarcia) [reply .travis.yml]: done

**Comment by fjrojasgarcia** [comment]
All suggestions done. Please review again.

File: test/qweb3.js (hunk 8 above)
Comment (fjrojasgarcia) [reply test/qweb3.js]: Adapted. Done.

File: test/qweb3.js (hunk 9 above)
Comment (fjrojasgarcia) [reply test/qweb3.js]: Done.

File: test/qweb3.js (hunk 10 above)
Comment (fjrojasgarcia) [reply test/qweb3.js]: Corrected.

File: test/qweb3.js (hunk 11 above)
Comment (fjrojasgarcia) [reply test/qweb3.js]: Adapted. Done.

File: test/qweb3.js (hunk 12 above)
Comment (fjrojasgarcia) [reply test/qweb3.js]: Same here. Done.

**Comment by dwalintukan** [comment]
Great job! Thanks for the contribution!

**Comment by fjrojasgarcia** [comment]
My pleasure.

**Pull request merged**